### DISCUSSION TITLE
Using renderRoot with react-router NavLink

### DISCUSSION THREAD

**Discussion by paynejj** [mantinedev/mantine#4950] [Q&A] [answered]
I am trying to use a react router NavLink in a mantine UnstyledButton via renderRoot so I can change styling based on the active page. Unlike the example on the release page, the UnstyledButton isn't rendering children or overriding the NavLink styles. If I don't put text in the NavLink, nothing renders. This did what I wanted with component={Link} and useLocation(). Are component and renderRoot not equivalent? Thanks!
```tsx
// NavBar.tsx
      <Group ml="xl" visibleFrom="sm">
        {navData.map((link) => (
          <UnstyledButton
            key={link.label}
            renderRoot={(_className, ...others) => (
              <NavLink
                style={({ isActive }) => ({
                  color: isActive ? "white" : "orange",
                })}
                to={link.link}
                {...others}
              >
                {link.label} // This text is rendered
              </NavLink>
            )}
          >
            {link.label} // This text is not rendered
          </UnstyledButton>
        ))}
      </Group>
```

**Comment by rtivital** [comment]
Because you override `children` prop passed to the component thought `...others` by setting `children` prop on NavLink.

**Reply by paynejj** [reply]
nothing renders without the child on Navlink. The below code renders nothing.
```tsx
<Group ml="xl" visibleFrom="sm">
        {navData.map((link) => (
          <UnstyledButton
            key={link.label}
            renderRoot={(_className, ...others) => (
              <NavLink
                style={({ isActive }) => ({
                  color: isActive ? "white" : "orange",
                })}
                to={link.link}
                {...others}
              />
            )}
          >
            {link.label}
          </UnstyledButton>
        ))}
      </Group>
```

**Reply by rtivital** [reply, marked as answer]
Because you did not destructure props correctly, `...others` is not the second argument of the function.
It should be: `renderRoot={({ className, ...others }) => (`

**Reply by paynejj** [reply]
Thank you!
